File: DSA/PriyankaSingh_590016827/day36question3.c

#include <stdio.h>

#define MAX 100

int stack[MAX];
int top = -1;

void push(int x) {
    stack[++top] = x;
}

int pop() {
    return stack[top--];
}

int isEmpty() {
    return top == -1;
}

void insertAtBottom(int x) {
    if (isEmpty()) {
        push(x);
    } else {
        int temp = pop();
        insertAtBottom(x);
        push(temp);
    }
}

void printStack() {
    for (int i = 0; i <= top; i++)
        printf("%d ", stack[i]);
    printf("\n");
}

int main() {
    push(1);
    push(2);
    push(3);

    int x = 0;
    insertAtBottom(x);

    printf("Stack after inserting at bottom: ");
    printStack();

    return 0;
}

File: DSA/LakshaKundra_590013237/Day_36_Question_3.py
def insert_at_bottom(stack, x):
    if not stack:
        stack.append(x)
    else:
        top = stack.pop()
        insert_at_bottom(stack, x)
        stack.append(top)

stack1 = [1, 2, 3]
insert_at_bottom(stack1, 0)
print(stack1)

stack2 = [5, 6]
insert_at_bottom(stack2, 9)
print(stack2)
